src/api_server/src/lib.rs

```diff
@@ -90,10 +91,10 @@ impl ApiServer {
         let sessions = web::Data::new(SessionMap::new());
         let http_timeout = self.http_timeout;
 
-        let repository = self
-            .config
-            .repository_type
-            .to_repository(&self.config.repository_description)?;
+        let repository = RepositoryType::from_name(
```

Comment:
Why can't we directly store the instanced repository instead of `RepositoryType`?  I really think the original plan is better.